File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # put input into a set
        # loop thru using while loop
        # if 1 less exists 

        nums = set(nums)
        longest_sequence = 0
        for num in nums.copy():
            sequence = 0
            while num - 1 not in nums and num in nums: 
                sequence += 1
                nums.remove(num)
                num += 1
            longest_sequence = max(longest_sequence, sequence)

        return longest_sequence

File: Data Structures & Algorithms/jump-game/submission-2.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        int farthestJump = 0;
        for (int i = 0; i < nums.size(); i++){
            if ( farthestJump >= i ){
                farthestJump = std::max(farthestJump, i + nums[i]);
            }
        }
/*
        
*/
        return (farthestJump >= nums.size() - 1);

    }
};
